src/types/onyx/OriginalMessage.ts

```diff
@@ -221,192 +130,178 @@ type ChronosOOOEvent = {
+    [CONST.REPORT.ACTIONS.TYPE.IOU]: OriginalMessageIOU;
+    [CONST.REPORT.ACTIONS.TYPE.MANAGER_ATTACH_RECEIPT]: never;
+    [CONST.REPORT.ACTIONS.TYPE.MANAGER_DETACH_RECEIPT]: never;
+    [CONST.REPORT.ACTIONS.TYPE.MARK_REIMBURSED_FROM_INTEGRATION]: never;
+    [CONST.REPORT.ACTIONS.TYPE.MARKED_REIMBURSED]: never;
+    [CONST.REPORT.ACTIONS.TYPE.MERGED_WITH_CASH_TRANSACTION]: never;
+    [CONST.REPORT.ACTIONS.TYPE.MODIFIED_EXPENSE]: OriginalMessageModifiedExpense;
+    [CONST.REPORT.ACTIONS.TYPE.MOVED]: OriginalMessageMoved;
+    [CONST.REPORT.ACTIONS.TYPE.OUTDATED_BANK_ACCOUNT]: never;
+    [CONST.REPORT.ACTIONS.TYPE.REIMBURSEMENT_ACH_BOUNCE]: never;
+    [CONST.REPORT.ACTIONS.TYPE.REIMBURSEMENT_ACH_CANCELLED]: never;
+    [CONST.REPORT.ACTIONS.TYPE.REIMBURSEMENT_ACCOUNT_CHANGED]: never;
+    [CONST.REPORT.ACTIONS.TYPE.REIMBURSEMENT_DEQUEUED]: OriginalMessageReimbursementDequeued;
+    [CONST.REPORT.ACTIONS.TYPE.REIMBURSEMENT_DELAYED]: never;
+    [CONST.REPORT.ACTIONS.TYPE.REIMBURSEMENT_QUEUED]: OriginalMessageReimbursementQueued;
+    [CONST.REPORT.ACTIONS.TYPE.REIMBURSEMENT_REQUESTED]: never;
+    [CONST.REPORT.ACTIONS.TYPE.REIMBURSEMENT_SETUP]: never;
+    [CONST.REPORT.ACTIONS.TYPE.RENAMED]: OriginalMessageRenamed;
+    [CONST.REPORT.ACTIONS.TYPE.REPORT_PREVIEW]: OriginalMessageReportPreview;
+    [CONST.REPORT.ACTIONS.TYPE.SELECTED_FOR_RANDOM_AUDIT]: never;
+    [CONST.REPORT.ACTIONS.TYPE.SHARE]: never;
+    [CONST.REPORT.ACTIONS.TYPE.STRIPE_PAID]: never;
+    [CONST.REPORT.ACTIONS.TYPE.SUBMITTED]: never;
+    [CONST.REPORT.ACTIONS.TYPE.TASK_CANCELLED]: never;
+    [CONST.REPORT.ACTIONS.TYPE.TASK_COMPLETED]: never;
+    [CONST.REPORT.ACTIONS.TYPE.TASK_EDITED]: never;
+    [CONST.REPORT.ACTIONS.TYPE.TASK_REOPENED]: never;
+    [CONST.REPORT.ACTIONS.TYPE.TAKE_CONTROL]: never;
+    [CONST.REPORT.ACTIONS.TYPE.UNAPPROVED]: never;
+    [CONST.REPORT.ACTIONS.TYPE.UNHOLD]: never;
+    [CONST.REPORT.ACTIONS.TYPE.UNSHARE]: never;
+    [CONST.REPORT.ACTIONS.TYPE.UPDATE_GROUP_CHAT_MEMBER_ROLE]: never;
+    [CONST.REPORT.ACTIONS.TYPE.REIMBURSEMENT_SETUP_REQUESTED]: never;
+    [CONST.REPORT.ACTIONS.TYPE.TRIPPREVIEW]: OriginalMessageTripRoomPreview;
+} & {
+    [T in ValueOf<typeof CONST.REPORT.ACTIONS.TYPE.POLICY_CHANGE_LOG>]: OriginalMessageChangeLog;
+} & {
+    [T in ValueOf<typeof CONST.REPORT.ACTIONS.TYPE.ROOM_CHANGE_LOG>]: OriginalMessageChangeLog;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+type AssertOriginalMessageDefinedForAllActions = AssertTypesEqual<
+    ReportActionName,
+    keyof OriginalMessageMap,
+    `Error: Types don't match, OriginalMessageMap type is missing: ${Exclude<ReportActionName, keyof OriginalMessageMap>}`
+>;
+
+type OriginalMessage<T extends ReportActionName> = OriginalMessageMap[T];
+
+// Note: type-fest's ConditionalKeys does not work correctly with objects containing `never`: https://github.com/sindresorhus/type-fest/issues/878
```

Comment:
This is outdated, I fixed it upstream and that was released in https://github.com/sindresorhus/type-fest/releases/tag/v4.18.3

So if you bump the `type-fest` version to latest, this can be simplified to:

```ts
import {ConditionalKeys} from 'type-fest';
type ReportActionNamesWithHTMLMessage = ConditionalKeys<OriginalMessageMap, {html: string}>;
```

but perhaps more importantly, `REPORT_ACTIONS_WITH_HTML_MESSAGE` is unused in your branch.